src/mono/mono.proj

```diff
@@ -724,9 +683,15 @@
       <MonoAotOffsetsPrefix Condition="'$(Platform)' == 'arm64'">$(MonoCrossDir)/usr/lib/gcc/aarch64-linux-gnu/5</MonoAotOffsetsPrefix>
     </PropertyGroup>
 
+    <PropertyGroup>
+      <_MonoLLVMTargetArchitecture>$(BuildArchitecture)</_MonoLLVMTargetArchitecture>
+      <_MonoLLVMTargetArchitecture Condition="'$(RealTargetArchitecture)' != ''">$(RealTargetArchitecture)</_MonoLLVMTargetArchitecture>
+    </PropertyGroup>
+
     <!-- macOS host specific options -->
     <ItemGroup Condition="'$(RealTargetOS)' == 'osx' or $([MSBuild]::IsOSPlatform('OSX'))">
-      <MonoAOTCMakeArgs Condition="'$(RealTargetArchitecture)' == 'arm64'" Include="-DCMAKE_OSX_ARCHITECTURES=arm64"/>
+      <MonoAOTCMakeArgs Condition="'$(_MonoLLVMTargetArchitecture)' == 'x64'" Include="-DCMAKE_OSX_ARCHITECTURES=x86_64" />
+      <MonoAOTCMakeArgs Condition="'$(_MonoLLVMTargetArchitecture)' != 'x64'" Include="-DCMAKE_OSX_ARCHITECTURES=$(_MonoLLVMTargetArchitecture)" />
```

Comment:
_MonoLLVMTargetArchitecture won't always have a valid value for CMAKE_OSX_ARCHITECTURES, e.g. x86 vs i386. You'll need separate checks for each arch value I think

Reply:
Sure, I can add those conditions (copy from above https://github.com/dotnet/runtime/blob/063adfe9f2925c5c6c74e5d30d04a7d5a085e144/src/mono/mono.proj#L471-L475), but I was under the impression that we don't really support / care for armv7 or i386 -darwin targets.

Reply:
that's a good point, we don't need those anymore but I still prefer the explicit mappings because it is easier to see what is going on.

we can remove the arm and x86 cases but we should add a check somewhere to prevent passing that in (can be done in a separate PR).